k8s/webhooks.go

```diff
@@ -0,0 +1,124 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/grafana/grafana-app-sdk/resource"
+)
+
+type WebhookServerConfig struct {
+	// The Port to run the HTTPS server on
+	Port int
+	// TLSConfig contains cert information for running the HTTPS server
+	TLSConfig TLSConfig
+	// ValidatingControllers is a map of schemas to their corresponding ValidatingAdmissionController.
+	ValidatingControllers map[resource.Schema]resource.ValidatingAdmissionController
+	// MutatingControllers is a map of schemas to their corresponding MutatingAdmissionController.
+	MutatingControllers map[resource.Schema]resource.MutatingAdmissionController
+	// DefaultValidatingController is called for any /validate requests received which don't have an entry in ValidatingControllers.
+	// If left nil, an error will be returned to the caller instead.
+	DefaultValidatingController resource.ValidatingAdmissionController
+	// DefaultMutatingController is called for any /validate requests received which don't have an entry in MutatingControllers.
+	// If left nil, an error will be returned to the caller instead.
+	DefaultMutatingController resource.MutatingAdmissionController
+	// WrapControllers instructs the WebhookServerController to wrap the provided Validating and Mutating AdmissionControllers
+	// in their Opinionated variants. This is done at setup time.
+	WrapControllers bool
+}
+
+type TLSConfig struct {
+	CertPath string
+	KeyPath  string
+}
+
+func NewWebhookServer(config WebhookServerConfig) (*WebhookServerController, error) {
```

Comment:
Yeah, the naming is a bit confusing because it was as I was waffling between the two possible new functions, and I had re-named the struct from `WebhookServer` to `WebhookServerController`. I'll make sure whatever new function seems best is named `NewWebhookServerController`.

I think your suggestion makes sense, thanks!